audioswitch/src/main/java/com/twilio/audioswitch/bluetooth/BluetoothHeadsetManager.kt

```diff
@@ -94,59 +99,46 @@ internal constructor(
-                            }
-                            AudioManager.SCO_AUDIO_STATE_DISCONNECTED -> {
-                                logger.d(TAG, "Bluetooth SCO Audio disconnected")
-                                /*
-                                 * This block is needed to restart bluetooth SCO in the event that
-                                 * the active bluetooth headset has changed.
-                                 */
-                                if (hasActiveHeadsetChanged()) {
-                                    enableBluetoothScoJob.executeBluetoothScoJob()
-                                }
-
-                                headsetListener?.onBluetoothHeadsetStateChanged()
-                                disableBluetoothScoJob.cancelBluetoothScoJob()
-                            }
-                            AudioManager.SCO_AUDIO_STATE_ERROR -> {
-                                logger.e(TAG, "Error retrieving Bluetooth SCO Audio state")
+        if (isCorrectIntentAction(intent.action)) {
+            intent.getHeadsetDevice()?.let { bluetoothDevice ->
+                intent.getIntExtra(BluetoothHeadset.EXTRA_STATE, STATE_DISCONNECTED).let { state ->
+                    when (state) {
+                        STATE_CONNECTED -> {
+                            logger.d(
+                                    TAG,
+                                    "Bluetooth headset $bluetoothDevice connected")
+                            connect()
+                            headsetListener?.onBluetoothHeadsetStateChanged(bluetoothDevice.name)
+                        }
+                        STATE_DISCONNECTED -> {
+                            logger.d(
+                                    TAG,
+                                    "Bluetooth headset $bluetoothDevice disconnected")
+                            disconnect()
+                            headsetListener?.onBluetoothHeadsetStateChanged()
+                        }
+                        STATE_AUDIO_CONNECTED -> {
+                            logger.d(TAG, "Bluetooth audio connected on device $bluetoothDevice")
+                            enableBluetoothScoJob.cancelBluetoothScoJob()
+                            headsetState = AudioActivated
+                            headsetListener?.onBluetoothHeadsetStateChanged()
+                        }
+                        STATE_AUDIO_DISCONNECTED -> {
+                            logger.d(TAG, "Bluetooth audio disconnected on device $bluetoothDevice")
+                            disableBluetoothScoJob.cancelBluetoothScoJob()
+                            /*
+                             * This block is needed to restart bluetooth SCO in the event that
+                             * the active bluetooth headset has changed.
+                             */
+                            if (hasActiveHeadsetChanged()) {
+                                enableBluetoothScoJob.executeBluetoothScoJob()
```

Comment:
Maybe this can be removed.

Reply:
@idelgado Just tested removing this change but unfortunately it is still needed even with these new changes :/